diff --git a/src/index.js b/src/index.js
index 66ae12c6f751f6e400cd49fb531b33985e915c02..814e7234b46ed17aafd57d60b3a4014adc997c07 100644
--- a/src/index.js
+++ b/src/index.js
@@ -28,7 +28,7 @@ module.exports = {
   extends: [
     'plugin:@typescript-eslint/recommended',
     'plugin:@typescript-eslint/recommended-requiring-type-checking',
-    'plugin:import/typescript',
+    'plugin:import-x/typescript',
   ],
 
   rules: {
@@ -36,7 +36,7 @@ module.exports = {
     '@typescript-eslint/no-extra-semi': 'off',
 
     // Handled by TypeScript
-    'import/no-unresolved': 'off',
+    'import-x/no-unresolved': 'off',
 
     // Our rules
     '@typescript-eslint/array-type': 'error',
